akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -178,10 +178,18 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
 
             case SettingsFrame(settings) ⇒
               if (settings.nonEmpty) debug(s"Got ${settings.length} settings!")
+
+              var settingsAppliedOk = true
+
               settings.foreach {
                 case Setting(Http2Protocol.SettingIdentifier.SETTINGS_INITIAL_WINDOW_SIZE, value) ⇒
-                  debug(s"Setting initial window to $value")
-                  multiplexer.updateDefaultWindow(value)
+                  if (value <= Http2Protocol.MaxInitialWindowSize) {
```

Comment:
Yes, I guess so. I thought we had precedent of checking other settings in the parser but it seems we don't. Go for it ;)